schematic/schemas/data_model_parser.py

```diff
@@ -0,0 +1,319 @@
+import logging
+import pandas as pd
+import pathlib
+from typing import Any, Dict, Optional, Text, List
+
+from schematic.utils.df_utils import load_df
+from schematic.utils.io_utils import load_json
+
+from schematic.schemas.data_model_relationships import (
+    DataModelRelationships
+    )
+
+from schematic import LOADER
+
+logger = logging.getLogger("Synapse storage")
+
+class DataModelParser():
+    '''
+    This class takes in a path to a data model and will convert it to an
+    attributes:relationship dictionarythat can then be further converted into a graph data model. 
+    Other data model types may be added in the future.
+    '''
+    def __init__(
+        self,
+        path_to_data_model: str,
+        ) -> None:
+        """
+        Args:
+            path_to_data_model, str: path to data model.
+        """
+
+        self.path_to_data_model = path_to_data_model
+        self.model_type = self.get_model_type(path_to_data_model)
+        self.base_schema_path = None
+
+    def _get_base_schema_path(self, base_schema: str = None) -> str:
+        """Evaluate path to base schema.
+
+        Args:
+            base_schema: Path to base data model. BioThings data model is loaded by default.
+
+        Returns:
+            base_schema_path: Path to base schema based on provided argument.
+        """
+        biothings_schema_path = LOADER.filename("data_models/biothings.model.jsonld")
+        self.base_schema_path = biothings_schema_path if base_schema is None else base_schema
+
+        return self.base_schema_path
+
+    def get_model_type(self, path_to_data_model: str) -> str:
+        '''Parses the path to the data model to extract the extension and determine the data model type.
+        Args:
+            path_to_data_model, str: path to data model
+        Returns:
+            str: uppercase, data model file extension.
+        Note: Consider moving this to Utils.
+        '''
+        return pathlib.Path(path_to_data_model).suffix.replace('.', '').upper()
+
+    def parse_base_model(self):
```

Comment:
The output here could be typed as `dict` at least. Maybe basemodel should be its own dataclass?